sdcm/utils/monitorstack.py

```diff
@@ -0,0 +1,251 @@
+                                                           dst_dir=monitor_stack_base_dir)
+    monitor_data_arch = extract_monitor_data_archive(downloaded_monitor_archive,
+                                                     monitor_stack_base_dir)
+    monitor_stack_arch = extract_monitor_stack_archive(downloaded_monitor_archive,
+                                                       monitor_stack_base_dir)
+
+    if not monitor_data_arch:
+        LOGGER.error("No prometheus snapshot were found in arch %s", arch['file_path'])
+        return False
+    if not monitor_stack_arch:
+        LOGGER.error("No monitor stack archive were found in arch %s", arch['file_path'])
+        return False
+
+    monitor_data_dir = create_monitoring_data_folder(monitor_stack_base_dir, monitor_data_arch)
+    monitor_stack_dir = create_monitoring_stack_folder(monitor_stack_base_dir, monitor_stack_arch)
+
+    if not monitor_stack_dir or not monitor_data_dir:
+        LOGGER.error('Creating monitor stack directories failed:\ndata_dir: %s; stack_dir: %s',
+                     monitor_data_dir, monitor_stack_dir)
+    _, scylla_version = get_monitorstack_scylla_version(monitor_stack_dir)
+
+    status = start_dockers(monitor_stack_dir,
+                           monitor_data_dir,
+                           scylla_version)
+    if status:
+        upload_sct_dashboards(monitor_stack_dir, scylla_version)
+        upload_annotations(monitor_stack_dir)
+        return status
+    else:
+        LOGGER.error('Error during dockers starting. Trying next arhive')
+        remove_files(monitor_stack_base_dir)
+        return False
+
+
+def get_monitor_stack_archive(archives, date_time):
+    arch = None
+    if not archives:
+        LOGGER.warning('No any archive for monitoring stack')
+    if date_time:
+        found_archives = [archive for archive in archives
+                          if archive['date'] == datetime.datetime.strptime(date_time, "%Y%m%d_%H%M%S")]
+        if not found_archives:
+            LOGGER.warning('Monitor stack archive for date %s was not found', date_time)
+        else:
+            arch = found_archives[0]
+    else:
+        arch = archives[0]
+
+    return arch
+
+
+def create_monitoring_data_folder(base_dir, archive):
+    monitor_data_base_dir = os.path.join(base_dir, 'monitor_data_dir')
+    cmd = dedent("""
+        mkdir -p {data_dir}
+        cd {data_dir}
+        cp {archive} ./
+        tar -xvf {archive_name}
+        chmod -R 777 {data_dir}
+        """.format(data_dir=monitor_data_base_dir,
+                   archive=archive,
+                   archive_name=os.path.basename(archive)))
+    result = LocalCmdRunner().run(cmd, ignore_status=True)
+    if result.exited > 0:
+        LOGGER.error("Error during extracting prometheus snapshot. Switch to next archive")
+        return False
+    return get_monitor_data_dir(monitor_data_base_dir)
+
+
+def create_monitoring_stack_folder(base_dir, archive):
```

Comment:
folders are in MS Windows, in Linux they are called directories
 `create_monitoring_stack_folder` -> `create_monitoring_stack_dir`

Reply:
renamed